layers/core/filter.py

```diff
@@ -59,7 +46,25 @@ def get_dict(self):
             if entry == 'domain':
                 continue
             if listing[entry] != UNSETVALUE:
-                temp[entry.split(type(self).__name__ + '__')[-1]] \
-                    = listing[entry]
+                subname = entry.split('__')[-1]
+                if subname != 'stages':
+                    temp[subname] = listing[entry]
         if len(temp) > 0:
             return temp
+
+class Filter(Filterv4):
```

Comment:
Why does class Filter inherit from Filterv4? Shouldn't the specific version of the filter object logically inherit from the generic filter (in other words, shouldn't it be the other way around)?

Reply:
It's a poor choice of names, but a side effect of how v4 has less fields than the v3 version of the object. I'll change things so that the default one is v4, and the extended is v3 to make things a bit more clear semantically.